docs/src/main/asciidoc/openapi-swaggerui-guide.adoc

```diff
@@ -284,6 +284,93 @@ quarkus.smallrye-openapi.path=/swagger
 
 Hit `CTRL+C` to stop the application.
 
+== Loading OpenAPI Schama From Static Files
+
+Instead of dynamically creating OpenAPI schemas from annotation scanning, Quarkus also proposes serving static OpenAPI documents.
+The static file to serve must be a valid document conforming to the https://swagger.io/docs/specification[OpenAPI specification].
+An OpenAPI document that conforms to the  OpenAPI Specification is itself a valid JSON object, that can be represented in
+a `yaml` or `json` formats.
+
+To see this in action, we'll create a static OpenAPI document `META-INF/openapi.yaml` for our `/fruits` path.
+You can choose a different document path as indicating in the <<open-document-paths, document paths tip below>>
+
+[source, shell]
+----
+openapi: 3.0.1
+info:
+  title: Static OpenAPI document of fruits resource
+  description: Fruit resources Open API documentation
+  version: "1.0"
+
+servers:
+  - url: http://localhost:8080/openapi
+    description: Optional dev mode server description
+
+paths:
+  /fruits:
+    get:
+      responses:
+        200:
+          description: OK - fruits list
+          content:
+            application/json: {}
+    post:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Fruit'
+      responses:
+        200:
+          description: new fruit resource created
+          content:
+            application/json: {}
+    delete:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Fruit'
+      responses:
+        200:
+          description: OK - fruit resource deleted
+          content:
+            application/json: {}
+components:
+  schemas:
+    Fruit:
+      properties:
+        description:
+          type: string
+        name:
+          type: string
+----
+
+Now, a request to `/openapi` endpoint will serve the static OpenAPI document instead of the generated one.
+
+[[open-document-paths]]
+[TIP]
+.About various OpenAPI document paths
+====
+Quarkus supports various file paths for the OpenAPI document. These are:
+
+* `META-INF/openapi.yaml`
+* `META-INF/openapi.yml`
+* `META-INF/openapi.json`
+* `WEB-INF/classes/META-INF/openapi.yml`
+* `WEB-INF/classes/META-INF/openapi.yaml`
+* `WEB-INF/classes/META-INF/openapi.json`
+
+Live reload of static OpenAPI document is supported during development. A modification to your OpenAPI document will be picked up on fly by Quarkus.
+====
+
+[TIP]
+.Disabling annotation scanning
+====
+When serving OpenAPI document from a static file, it is interesting to disable the dynamic generation of OpenAPI document through annotation scanning which will otherwise ovverride your static OpenAPI document.
+To do this, add the `mp.openapi.scan.disable=true` configuration to your `application.properties` file.
```

Comment:
Interesting, I wonder if the presence of a static file should just force the scanning to be disabled? Thoughts?

Reply:
Also asked on the quarkus-dev mailing list

Reply:
Then this will force the check of the presence of "any" of the static files. Having a configuration that's document is enough IMO. 

I wonder what others think? @kenfinnigan @gsmet

Reply:
> Also asked on the quarkus-dev mailing list

My page was not in sync so obvious I did not see this. 

Great. Let's wait for response.

Reply:
Scanning isn't disabled because it combines the two models.

Currently you need to manually disable scanning with config